### DISCUSSION TITLE
Is it possible to let one virtual node associate with python scripts to achieve some functions?

### DISCUSSION THREAD

**Discussion by GitZinger** [shadow/shadow#760] [Q&A] [answered]
**State your question concisely**

Is it possible to let one virtual node associate with one python script to achieve some functions?

for each bridge node or client or server node, can I add some python scripts solely individually to achieve some goals? along with Tor and TGen together?
in detail, the python scripts could take in effect in the layer between network and Tor?

**Describe your current understanding of Shadow as it relates to your question**

I am completely new to here

**Describe which parts you are unsure about**

**Comment by robgjansen** [comment, marked as answer]
With the current version of Shadow, it will be difficult to run python scripts in the Shadow virtual hosts. We do not support this method right now, but we will eventually in Shadow v2. The closest we have right now is the python plugin that was written by another contributor, but again, I don't think this is working correctly:
https://github.com/shadow/shadow-plugin-extras/tree/master/python
